api/blogs/controllers/blogs.js

```diff
@@ -5,4 +5,124 @@
  * to customize this controller
  */
 
-module.exports = {};
+const validator = require('validator');
+const { sanitizeEntity } = require("strapi-utils");
+
+module.exports = {
+  async startBlogPost(ctx) {
+    try {
+      const query = ctx.query;
+      if(validator.isLength(query.title, {min:1, max: 255}) && validator.isLength(query.content, {min:1})) {
+        const blog = await strapi.services.blogs.create({
+          title: query.title,
+          content: query.content,
+          author: ctx.req.user.id
+        })
+        const sanitizedBlog = sanitizeEntity(blog, {
+          model: strapi.models["blogs"],
+        });
+        ctx.send({
+          success: true,
+          id: sanitizedBlog.id
+        });
+      } else {
+        ctx.send({
+          success: false,
+          reason: "Title (up to 255 chars) and Content are required."
+        });
+      }
+    } catch (error) {
+      ctx.send({
+        success: false,
+        reason: "Internal Error."
+      });
+    }
+  },
+
+  async getBlogPost(ctx) {
+    try {
+      const query = ctx.query;
+      if(validator.isLength(query.id, {min:1})) {
+        const blog = await strapi.services.blogs.findOne({
+          id: query.id,
+        })
+        const sanitizedBlog = sanitizeEntity(blog, {
+          model: strapi.models["blogs"],
+        });
+        ctx.send({
+          success: false,
+          blog: {
+            id: sanitizedBlog.id,
+            title: sanitizedBlog.title,
+            summary: sanitizedBlog.content,
+            author: {
+              id: sanitizedBlog.author.id,
+              username: sanitizedBlog.author.username,
+            },
+            start_time: sanitizedBlog.created_at,
+            is_draft: sanitizedBlog.is_draft,
+            last_edit: sanitizedBlog.updated_at,
+            publish_time: sanitizedBlog.published_at
+          }
+        });
+      } else {
+        ctx.send({
+          success: false,
+          reason: "Title (up to 255 chars) and Content are required."
+        });
+      }
+    } catch (error) {
+      ctx.send({
+        success: false,
+        reason: "Internal Error."
+      });
+    }
+  },
+
+  async getBlogPosts(ctx) {
+    try {
+      const query = ctx.query;
```

Comment:
just to be clear, we are not designing this to be sqlite compliant only, right? This would still work with mongo without a single change in the code , right?